src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -81,6 +86,30 @@ public JsonWebToken(string jwtEncodedString)
             ReadToken(jwtEncodedString);
         }
 
+#if NET6_0_OR_GREATER
+        /// <summary>
+        /// Initializes a new instance of <see cref="JsonWebToken"/> from a span in JWS or JWE Compact serialized format.
+        /// </summary>
+        /// <param name="encodedTokenMemory">A span containing the JSON Web Token serialized in JWS or JWE Compact format.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="encodedTokenMemory"/> is empty.</exception>
+        /// <exception cref="ArgumentException"><paramref name="encodedTokenMemory"/> does not represent a valid JWS or JWE Compact serialization format.</exception>
+        /// <remarks>
+        /// See: https://datatracker.ietf.org/doc/html/rfc7519 (JWT)
+        /// See: https://datatracker.ietf.org/doc/html/rfc7515 (JWS)
+        /// See: https://datatracker.ietf.org/doc/html/rfc7516 (JWE)
+        /// <para>
+        /// The contents of the returned <see cref="JsonWebToken"/> have not been validated; the JSON Web Token is simply decoded. Validation can be performed using the methods in <see cref="JsonWebTokenHandler"/>.
+        /// </para>
+        /// </remarks>
+        public JsonWebToken(ReadOnlyMemory<char> encodedTokenMemory)
```

Comment:
>ReadOnlyMemory [](http://example.com/codeflow?start=28&length=14)

this will help ensure the memory is not accidently modified, correct?

Reply:
Correct. ReadyOnlyMemory is recommended when working with memory that we don't intend to modify.

Reply:
we may want to talk to partners about expectations here.

I'm not convinced we should be using Memory<T> over Span<T> here.